### ISSUE TITLE
Linting inside Markdown blocks outside of .md files

### ISSUE THREAD

**Issue by lleger** [DavidAnson/vscode-markdownlint#25]
Is it possible to get linting inside of Markdown-like blocks inside of non-Markdown files? The use case here is in [Elixir documentation](https://hexdocs.pm/elixir/writing-documentation.html), which is formatted in Markdown, but it’s actually just a block inside of a `.ex` or `.exs` file.

From the above link, an example:
```elixir
defmodule MyApp.Hello do
  @moduledoc """
  This is the Hello module.
  """

  @doc """
  Says hello to the given `name`.

  Returns `:ok`.

  ## Examples

      iex> MyApp.Hello.world(:john)
      :ok

  """
  def world(name) do
    IO.puts "hello #{name}"
  end
end
```

What’s between the `"""` of `@moduledoc` and `@doc` (and also, though not shown, `@typedoc`) is formatted with Markdown. Getting the warnings and/or, if possible, the auto-formatting inside of those blocks would be tremendous!

**Comment by DavidAnson** [comment]
Does VS Code recognize that section as Markdown? That’s the first step.

**Comment by lleger** [comment]
How be happy to check that for you — how does VSCode determine that, and how would I verify?

**Comment by DavidAnson** [comment]
Open the file in Code and see if you get Markdown intellisense in that block. Maybe also check if the language chooser in the lower right changes.

More info: https://code.visualstudio.com/docs/editor/intellisense

**Comment by lleger** [comment]
As far as I can tell, it doesn’t seem to be picking it up.

**Comment by DavidAnson** [comment]
Bummer. That’s definitely how I’d want to go about this. I’ll leave the issue open, but please don’t hold your breath. :)

**Comment by lleger** [comment]
Is that something that would have to be added upstream in VSCode itself in order to hook into it?

**Comment by DavidAnson** [comment]
That would be ideal, yes. If that support were present (and it might be, I just didn’t see it from my phone) then an extension could easily find out which parts of a document it can understand and could then process them. Without support like that, each extension needs to understand every relevant file type, parse it, and find the relevant sections. It’s possible, but it’s a lot more work and it doesn’t scale as well.

**Comment by lleger** [comment]
So, I think it must be possible already, since it works with HTML/JS. Test: create a file, set language to HTML. Add inline `script` tags. Type some JS—intellisense is there.

Google fu is failing me though, haven’t been able to figure out what it’s called in order to track down docs.

<img width="723" alt="screen shot 2018-01-16 at 10 53 22 pm" src="https://user-images.githubusercontent.com/52359/35026351-319b9fc6-fb10-11e7-9e92-5935bd97c542.png">

**Comment by lleger** [comment]
I think this plugin does something similar: provides HTML intellisense inside of JS files. https://github.com/Microsoft/typescript-lit-html-plugin

**Comment by texastoland** [comment]
Here's a grammar injection approach https://github.com/mjbvz/vscode-comment-tagged-templates.
